### PR TITLE
[New Integration] Threat Map

### PR DESCRIPTION
**Pull request by navnit-crest**
## Proposed commit message
Create a new integration package named threat_map and add the Threat Map dashboard.

The Threat Map dashboard includes a custom map visualization that uses the Elastic Map Service to add a base map. For the marks (path, text, arc, etc.) and animations, it uses the [Vega](https://vega.github.io/vega/) visualization framework within Kibana.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

1. Clone integrations repo.
2. Install elastic package locally.
3. Start elastic stack using elastic-package.
4. Move to integrations/packages/threat_map directory.
5. Run the following command to run tests. 
> elastic-package test

```
--- Test results for package: threat_map - START ---
╭────────────┬─────────────┬───────────┬─────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                           │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼─────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ threat_map │             │ asset     │ dashboard threat_map-7c35935c-0bb9-493a-ab49-d89889a5412d is loaded │ PASS   │      1.005µs │
╰────────────┴─────────────┴───────────┴─────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: threat_map - END   ---
Done
```
## Screens
threat-map

### CODE REVIEW

File: packages/missile_map/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+| `destination_label`           | Label at the destination location                                  | Optional          |               |
+| `pulse_at_source`             | If true, the pulse begins at the source instead of the destination | Optional          | `false`       |
+
+---
+
+## Usage
+
+The Missile Map visualization can be added to other dashboards in two ways:
+
+### 1. Duplicate the Entire Dashboard
```
Comment (kcreddy) [inline packages/missile_map/docs/README.md:47]:
```suggestion
### 1. Duplicate the Dashboard
```

File: packages/missile_map/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+The **Missile Map** dashboard visualizes network traffic flow between regions using animated paths. It includes missile-like arrows indicating direction and pulsing arcs at the destination.
+
+> **Note:** The animations may increase browser CPU usage.
+
+---
+
+## Prerequisites
+
+To use the Missile Map visualization, ensure the following:
```
Comment (kcreddy) [inline packages/missile_map/docs/README.md:11]:
```suggestion
To use the Missile Map dashboard, ensure the following:
```

File: packages/missile_map/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+policy_templates:
+  - name: sample
+    title: Sample logs
+    description: Collect sample logs
+    inputs:
+      - type: logfile
+        title: Collect sample logs from instances
+        description: Collecting sample logs
+owner:
+  github: elastic/integrations
```
Comment (kcreddy) [inline packages/missile_map/manifest.yml:35]:
```suggestion
  github: elastic/security-service-integrations
```

File: packages/missile_map/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+version: 0.1.0
+source:
+  license: "Elastic-2.0"
+description: "This integration ships a missile map, a custom visualization to analyze network traffic for Elasticsearch data."
+type: integration
+categories:
+  - custom
+conditions:
+  kibana:
+    version: "^8.10.0"
```
[the remark this answers is not in the document]
Comment (navnit-crest) [reply packages/missile_map/manifest.yml:13]: We chose `^8.10.0` to support a wider range of versions that align with the requirements. However, we can update it to `8.13.0` if that is preferred.

[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/missile_map/manifest.yml:13]: @jamiehynds , could you confirm if we can change the minimum kibana version to `8.14.0`, as it is used by most new integrations we are building?

[the remark this answers is not in the document]
Comment (jamiehynds) [reply packages/missile_map/manifest.yml:13]: @kcreddy the customer behind this request is on `8.15.0` so no issues with setting the minimum to `8.14.0`

[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/missile_map/manifest.yml:13]: @navnit-crest , can you change this to `^8.14.0` and also build it on `8.14.0` elastic-package stack version.

Also, export dashboard once again on this version: `elastic-package export dashboards`

File: packages/missile_map/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+format_version: 3.3.0
+name: missile_map
+title: Missile Map
+version: 0.1.0
+description: "This integration ships a missile map, a custom visualization to analyze network traffic data."
```
Comment (kcreddy) [inline packages/missile_map/manifest.yml:5]: Package name and description requires a change. Holding off merge until then.

**Review by kcreddy — CHANGES_REQUESTED** [review]

File: packages/threat_map/docs/README.md
Comment (kcreddy) [inline packages/threat_map/docs/README.md:1]: This `README.md` should be under `_dev/build/docs/README.md`. [Example](https://github.com/elastic/integrations/blob/main/packages/abnormal_security/_dev/build/docs/README.md) 

When you run `build`, the `docs/README.md` is auto-generated.
Reference:
https://github.com/elastic/elastic-package/blob/main/docs/howto/add_package_readme.md

**Review by kcreddy — COMMENTED** [review]
Please update PR's `Proposed commit message` section with updated package name.

**Pull request merged**